corehq/apps/users/tests/test_signals.py

```diff
@@ -140,3 +142,26 @@ def simple_doc(user):
         with patch.object(user_adapter, 'from_python', simple_doc):
             update_user_in_es(None, user)
         self.assertTrue(user_adapter.exists(user.user_id))
+
+
+class TestDjangoUserPostSaveSignal(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain = 'django-user-test'
+        cls.domain_obj = create_domain(cls.domain)
+        cls.addClassCleanup(cls.domain_obj.delete)
+
+    def test_signal_can_handle_stale_django_user(self):
+        couch_user = WebUser.create(self.domain, 'test@example.com', '***', created_by=None, created_via=None)
+        stale_django_user = couch_user.get_django_user()
+        # make stale_django_user stale
+        couch_user.email = 'test@example.com'
```

Comment:
I was initially confused about how this made the user stale, and it took me a bit of digging to understand the relationship between `email` and `username`. It might make this test clearer if a different email (i.e., different from the username) was used here and/or if `email` (with a value different from `username`) was passed to `WebUser.create(...)` above.